### DISCUSSION TITLE
Upon changing a keybind it should not prompt you to change the next keybind.

### DISCUSSION THREAD

**Discussion by YotGun** [ppy/osu#29957] [Ideas]
After changing a keybind, it automatically starts to change the next keybind. Let's say I change the keybind for left button and press escape, it does not leave the menu but changes the keybind for Right button into escape instead. I don't think this is very intuitive.

Additional note; it only seems to happen with letters. F keys or mouse buttons do not reproduce this behaviour

**Comment by peppy** [comment]
The funny part is this was specifically changed this way to match stable and to make it easier to quickly bind multiple bindings.

I can agree with adding <kbd>Esc</kbd> as a special case handling here, to abort the operation. @bdach thoughts?

**Reply by bdach** [reply]
How would I bind any binding to <kbd>Esc</kbd> with that special case in place? That's not academic, the back binding is rebindable, and currently bound to <kbd>Esc</kbd>.

**Reply by peppy** [reply]
I was proposing it only works when in automatic advance mode, ie gameplay bindings where esc is never used.

**Reply by bdach** [reply]
Sure I guess, although that proposal feels a bit edge-casey and clunky from a UX standpoint (not very clear why Esc works in some places and not others).

**Reply by YotGun** [reply]
just remembered i had this discussion open. is there a reason why its not mimicking stable behaviour? because in stable after changing a keybind it doesnt prompt you to change the next one and esc cancels it. (image attached)
![{9C5C8E69-6309-43DB-BED9-427735D2F59F}](https://github.com/user-attachments/assets/6590e166-0f30-4448-88c7-6e2ab7dce217)
